src/sse/sse_prime_butterflies.rs

```diff
@@ -0,0 +1,15166 @@
+        let twiddle2im = unsafe { _mm_set_pd(tw2.im, tw2.im) };
+        let twiddle3re = unsafe { _mm_set_pd(tw3.re, tw3.re) };
+        let twiddle3im = unsafe { _mm_set_pd(tw3.im, tw3.im) };
+        let twiddle4re = unsafe { _mm_set_pd(tw4.re, tw4.re) };
+        let twiddle4im = unsafe { _mm_set_pd(tw4.im, tw4.im) };
+        let twiddle5re = unsafe { _mm_set_pd(tw5.re, tw5.re) };
+        let twiddle5im = unsafe { _mm_set_pd(tw5.im, tw5.im) };
+        let twiddle6re = unsafe { _mm_set_pd(tw6.re, tw6.re) };
+        let twiddle6im = unsafe { _mm_set_pd(tw6.im, tw6.im) };
+        let twiddle7re = unsafe { _mm_set_pd(tw7.re, tw7.re) };
+        let twiddle7im = unsafe { _mm_set_pd(tw7.im, tw7.im) };
+        let twiddle8re = unsafe { _mm_set_pd(tw8.re, tw8.re) };
+        let twiddle8im = unsafe { _mm_set_pd(tw8.im, tw8.im) };
+        let twiddle9re = unsafe { _mm_set_pd(tw9.re, tw9.re) };
+        let twiddle9im = unsafe { _mm_set_pd(tw9.im, tw9.im) };
+        let twiddle10re = unsafe { _mm_set_pd(tw10.re, tw10.re) };
+        let twiddle10im = unsafe { _mm_set_pd(tw10.im, tw10.im) };
+        let twiddle11re = unsafe { _mm_set_pd(tw11.re, tw11.re) };
+        let twiddle11im = unsafe { _mm_set_pd(tw11.im, tw11.im) };
+
+        Self {
+            direction,
+            _phantom: std::marker::PhantomData,
+            rotate,
+            twiddle1re,
+            twiddle1im,
+            twiddle2re,
+            twiddle2im,
+            twiddle3re,
+            twiddle3im,
+            twiddle4re,
+            twiddle4im,
+            twiddle5re,
+            twiddle5im,
+            twiddle6re,
+            twiddle6im,
+            twiddle7re,
+            twiddle7im,
+            twiddle8re,
+            twiddle8im,
+            twiddle9re,
+            twiddle9im,
+            twiddle10re,
+            twiddle10im,
+            twiddle11re,
+            twiddle11im,
+        }
+    }
+
+    #[inline(always)]
+    pub(crate) unsafe fn perform_fft_contiguous(
+        &self,
+        input: RawSlice<Complex<T>>,
+        output: RawSliceMut<Complex<T>>,
+    ) {
+        let v0 = _mm_loadu_pd(input.as_ptr() as *const f64);
+        let v1 = _mm_loadu_pd(input.as_ptr().add(1) as *const f64);
+        let v2 = _mm_loadu_pd(input.as_ptr().add(2) as *const f64);
+        let v3 = _mm_loadu_pd(input.as_ptr().add(3) as *const f64);
+        let v4 = _mm_loadu_pd(input.as_ptr().add(4) as *const f64);
+        let v5 = _mm_loadu_pd(input.as_ptr().add(5) as *const f64);
+        let v6 = _mm_loadu_pd(input.as_ptr().add(6) as *const f64);
+        let v7 = _mm_loadu_pd(input.as_ptr().add(7) as *const f64);
+        let v8 = _mm_loadu_pd(input.as_ptr().add(8) as *const f64);
+        let v9 = _mm_loadu_pd(input.as_ptr().add(9) as *const f64);
+        let v10 = _mm_loadu_pd(input.as_ptr().add(10) as *const f64);
+        let v11 = _mm_loadu_pd(input.as_ptr().add(11) as *const f64);
+        let v12 = _mm_loadu_pd
```

Comment:
Recommendation: You might see a sizeable speedup if you split these addition trees in half.

It takes 4 cycles for each addition to go through, and the cpu can process 2 additions per cycle, so 8 simultaneous additions/subtractions could be pipelined. But the way these are arranged, it has to wait for each to finish before starting the next.

In practice, I bet the instruction reorderer can help a little, but there are definitely more variables than there are registers here.

Another recommendation is to try computing `temp_b10_13` , `temp_b10_14` and all the others right before the beginning of this tree, to hopefully reduce stack thrashing.

Reply:
I'll try! I wrote this assuming the compiler would look at the code, frown a little, and just reorder the whole thing. But maybe it isn't as smart as I thought.

Reply:
Just tried this on the length 11 butterfly. I can't measure any difference, it seems to have had no effect at all. So the compiler really does seem clever.